### DISCUSSION TITLE
Feature Request: `list_count_matches`

### DISCUSSION THREAD

**Discussion by moritzwilksch** [duckdb/duckdb#11692] [Ideas]
It'd be great to have a shorthand for counting the number of occurrences of an element in a list, similar to polar's [`.list.count_matches`](https://docs.pola.rs/py-polars/html/reference/expressions/api/polars.Expr.list.count_matches.html#polars.Expr.list.count_matches).
One can build this in a somewhat verbose way like this:

```sql
SELECT len(list_filter(listcol, e -> e = 42)) as number_of_42s
FROM table
```

whereas something like

```sql
SELECT list_count_matches(listcol, 42) as number_of_42s
-- or:
-- SELECT list_count_matches(listcol, e -> e=42) as number_of_42s
FROM table
```

would be nice and parsimonious.

**Comment by soerenwolfers** [comment]
This would save you five characters at the cost of having to remember it's `list_count_matches`, not `list_count_equals` or  `list_count_element` or `list_number_matches`. Personally, I'd rather those characters were saved by making list functions available without the `list_`. 

Then again, this might make a great first PR, see https://github.com/duckdb/duckdb/commit/21c0a04b3c95cf7985d9fe3de87e7a4d7a2a5a43.  Disclaimer: I'm not a maintainer and can't predict whether the PR would be accepted.

**Reply by moritzwilksch** [reply]
Fair point. Are there any other performance considerations to be aware of? For example, I noticed that

```sql
SELECT list_transform(listcol, e -> (e = 42)::int).list_sum()
```

is significantly slower when selecting many columns this way. Maybe there is a more efficient way than calculating the length of the filtered list.

**Reply by soerenwolfers** [reply]
I bet there are. However, there are already a bunch of slow macros in there (e.g. `list_intersect` takes m*n when it could take m+n) so a correct order of magnitude solution would likely be good enough to start with.

**Reply by soerenwolfers** [reply]
btw, `list_count(list, lambda)` was previously suggested at https://github.com/duckdb/duckdb/issues/1940 Could do that and a `list_count(list, element)` for good measure, can't think of possible ambiguities right now

**Reply by soerenwolfers** [reply]
also, implementing this would reduce unnecessary allocation of the new list compared to the existing solutions (but who knows, maybe the C++ is smart enough to already avoid that)
